src/s3ql/umount.py

```diff
@@ -89,14 +89,27 @@ def get_cmdline(pid):
     cannot be determined for other reasons, log warning
     and return None.
     '''
+    
+    output = None
+    if os.path.isdir('/proc'):
+        cmd_path = '/proc/%d/cmdline' % pid
+        if os.path.isfile(cmd_path):
+            try:
+                with open(cmd_path, 'r') as cmd_file:
+                    output = cmd_file.read()
+
+            except Exception:
+                log.warning('Error when reading from procfs, assuming process %d has terminated.'
```

Comment:
This is racy. The process may terminate between the `isfile` check and the opening, then you'd still get a spurious error message. So I would drop the `isfile` check completely, catch only the `FileNotFound` exception and (when caught) return `None` without logging an error.

Reply:
Completely agree, thanks! Changed.